pkg/services/ngalert/state/state_tracker.go

```diff
@@ -54,36 +57,36 @@ func NewStateTracker(logger log.Logger) *StateTracker {
 	return tracker
 }
 
-func (st *StateTracker) getOrCreate(alertRule *ngModels.AlertRule, result eval.Result) AlertState {
+func (st *StateTracker) getOrCreate(alertRule *ngModels.AlertRule, result eval.Result, processingTime time.Duration) AlertState {
 	st.stateCache.mu.Lock()
 	defer st.stateCache.mu.Unlock()
-	lbs := data.Labels{}
-	if len(result.Instance) > 0 {
-		lbs = result.Instance
-	}
+
+	// if duplicate labels exist, alertRule label will take precedence
+	lbs := mergeLabels(alertRule.Labels, result.Instance)
 	lbs["__alert_rule_uid__"] = alertRule.UID
```

Comment:
I'm fine with these labels as an intermediate step - but we'll have to revisit this. I have a theory around what we could use but I need to verify that it is sound. 

Let's discuss it in the issue we have open for this. The big three for Prometheus are `instance`, `job` and `alertname` under regular operation with defaults these are the three labels that would identify where an alert is coming from quasi-uniquely.

We need to come up with something similar for Grafana. Title I believe is not unique, Datasources could be more than one,  I think namespace (the name not UID) + title is a good combination but there might be more...

Reply:
I'll resolve this so we can have the conversation there. I threw these out there as a first attempt to but something even remotely meaningful into SOME labels but I don't have strong opinions about what this label set should be, or be called